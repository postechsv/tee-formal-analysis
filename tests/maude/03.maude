set show timing off .
set show advisories off .

select STRING .

red ascii("A") .
red ascii("AB") .

red length("AAA") .
red length("") .
red length("abcdefghijklmnopqrstuvwxyz") .

red substr("abcdefghijklmnopqrstuvwxyz", 10, 5) .
red substr("abcdefghijklmnopqrstuvwxyz", 1000000000000000000, 5) .
red substr("abcdefghijklmnopqrstuvwxyz", 10, 500000000000000000) .