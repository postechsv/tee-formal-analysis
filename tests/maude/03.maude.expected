==========================================
reduce in STRING : ascii("A") .
rewrites: 1
result NzNat: 65
==========================================
reduce in STRING : ascii("AB") .
rewrites: 0
result [FindResult]: ascii("AB")
==========================================
reduce in STRING : length("AAA") .
rewrites: 1
result NzNat: 3
==========================================
reduce in STRING : length("") .
rewrites: 1
result Zero: 0
==========================================
reduce in STRING : length("abcdefghijklmnopqrstuvwxyz") .
rewrites: 1
result NzNat: 26
==========================================
reduce in STRING : substr("abcdefghijklmnopqrstuvwxyz", 10, 5) .
rewrites: 1
result String: "klmno"
==========================================
reduce in STRING : substr("abcdefghijklmnopqrstuvwxyz", 1000000000000000000, 5)
    .
rewrites: 1
result String: ""
==========================================
reduce in STRING : substr("abcdefghijklmnopqrstuvwxyz", 10, 500000000000000000)
    .
rewrites: 1
result String: "klmnopqrstuvwxyz"
Bye.
